force-app/main/default/classes/ADDR_Addresses_UTIL_TEST.cls

```diff
@@ -0,0 +1,1341 @@
+
+        Test.startTest();
+        Boolean isAccAddressEmpty = ADDR_Addresses_UTIL.isAccAddressEmpty(account);
+        Test.stopTest();
+
+        System.assertEquals(true, isAccAddressEmpty);
+    }
+
+    @isTest
+    static void testIsAccAddressEmptyNotNullNotEmptyFalse() {
+        Account account = UTIL_UnitTestData_TEST.getMultipleTestAccounts(1, UTIL_Describe_API.getBizAccRecTypeID())[0];
+        account.BillingStreet = 'Some random billing street';
+
+        Test.startTest();
+        Boolean isAccAddressEmpty = ADDR_Addresses_UTIL.isAccAddressEmpty(account);
+        Test.stopTest();
+
+        System.assertEquals(false, isAccAddressEmpty);
+    }
+
+    /**************************************************************************************************************************
+     ****************************************************** equalsCaseSensitive ***********************************************
+     **************************************************************************************************************************/
+    @isTest
+    static void testEqualsCaseSensitiveBothNullTrue() {
+        Test.startTest();
+        Boolean stringEquals = ADDR_Addresses_UTIL.equalsCaseSensitive(null, null);
+        Test.stopTest();
+
+        System.assertEquals(true, stringEquals);
+    }
+
+    @isTest
+    static void testEqualsCaseSensitiveOnlyStr1NullFalse() {
+        Test.startTest();
+        Boolean stringEquals = ADDR_Addresses_UTIL.equalsCaseSensitive(null, 'test string');
+        Test.stopTest();
+
+        System.assertEquals(false, stringEquals);
+    }
+
+    @isTest
+    static void testEqualsCaseSensitiveOnlyStr2NullFalse() {
+        Test.startTest();
+        Boolean stringEquals = ADDR_Addresses_UTIL.equalsCaseSensitive('test string', null);
+        Test.stopTest();
+
+        System.assertEquals(false, stringEquals);
+    }
+
+    @isTest
+    static void testEqualsCaseSensitiveNotNullEqualsTrue() {
+        Test.startTest();
+        Boolean stringEquals = ADDR_Addresses_UTIL.equalsCaseSensitive('test string', 'test string');
+        Test.stopTest();
+
+        System.assertEquals(true, stringEquals);
+    }
+
+    @isTest
+    static void testEqualsCaseSensitiveNotNullEqualsFalse() {
+        Test.startTest();
+        Boolean stringEquals = ADDR_Addresses_UTIL.equalsCaseSensitive('test string', '!test string');
+        Test.stopTest();
+
+        System.assertEquals(false, stringEquals);
+    }
+
+    /**************************************************************************************************************************
+     ****************************************************** isAddressMatchFuzzy ***********************************************
+     **************************************************************************************************************************/
+    //TODO: The method is not test visible.
```

Comment:
Making a method test visible is fine as it doesn't affect functionality outside of tests. Go ahead and do it then write the tests. This is especially important since this is one that's going to be critical for the upcoming work.

Reply:
On it.